src/event_handler.py

```diff
@@ -0,0 +1,40 @@
+import boto3
+import os
+from src.event_mapper import event_from_json
+
+
+# noinspection PyUnusedLocal
+def store_queued_events(_, __):
+    sqs_client = boto3.client('sqs')
+    queue_url = os.environ['QUEUE_URL']
+
+    while True:
+        message = __fetch_single_message(sqs_client, queue_url)
+        if message is None:
+            break
+
+        # noinspection PyBroadException
+        # catch all errors and log them - we never want a single failing message to kill the process.
+        try:
+            event = event_from_json(message['Body'])
+            print(event.event_id)
+            __delete_message(sqs_client, queue_url, message)
+        except Exception as exception:
+            print('Failed to store message - Exception: {0}'.format(str(message)))
+
+
+def __fetch_single_message(sqs_client, queue_url):
```

Comment:
More of a question than a comment - when does SQS mark a message as consumed?  It's important for our event to be sort-of transactional.  If we receive a message, then start writing it to the database, and then the database write fails, or the process crashes, or lightning hits the data center, I'd want the SQS message to stay on the queue and be (eventually) re-delivered.

Reply:
As I understand it, SQS won't mark it as consumed until we explicitly 'delete' the message from the queue. 
Deleting the message is the last thing we do, so any exception would mean that the message remains on the queue.

We have a test for this. When the message is invalid, an error is thrown, and we assert that there should be one (not visible) message remaining in the queue.